docs/changes.rst

```diff
@@ -22,6 +22,17 @@ You should generally assume that an API is internal unless you have specific
 information to the contrary.
 
 -------------------
+3.14.0 - 2017-07-20
```

Comment:
Keeping the current date in the changelog can be annoying, but it lets us turn merges immediately into releases!  See #555 for more on this (including discussion of how to automate the date part away).

I'd merge the paragraphs you've already got here, and add a reference to the issue and acknowledgement:

    This is a feature release.  The :func:`~hypothesis.strategies.characters` strategy now takes a ``whitelist_characters`` argument, which are added to the characters it will produce.  (:issue:`668`)
    Thanks to Moreati for this feature.

Plus - add youself to the contributors list in CONTRIBUTING.rst!  (Alphabetical, name or psudeonym, email optional)